### PR TITLE
Use extend arguments

### PR DESCRIPTION
**Pull request by joaomvfsantos**
I had the need not use "flatten". I think we should be able to pass extra arguments as a separate (optional) input parameter.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -54,12 +54,30 @@ exports.generateFdf = function(data) {
...
   var tempName       = temp.path({suffix: '.pdf'}),
       tempNameResult = temp.path({suffix: '.pdf'}),
       pdfPath        = isAbsolute(templatePath) ? templatePath : path.join(__dirname, templatePath);
 
-  child = spawn("pdftk", [pdfPath, "fill_form", "-", "output", tempName, "flatten"]);
+  // retrieve arguments as array
+  var args = [];
+  for (var i = 0; i < arguments.length; i++) {
+       args.push(arguments[i]);
+  }
```
Comment (ScottONeal) [inline index.js]: Anyway to just loop over arguments after data && templatePath? Would help in readability and you wouldn't have to have lines 68 & 69.

**Comment by ScottONeal** [comment]
Hey thanks for this PR! I think this would be a nice option to have.

In looking at your code, I think we can clean up some of the argument handling by doing some assertion tests instead of looping over all of the `arguments`. Something on the lines of:

```js
function(data, templatePath, extendArgs, callback) {

  // Check if extendArgs is our callback, adds backwards compat
  if ( typeof extendArgs === 'function' ) {
   callback = extendArgs;
  }
  else if ( typeof extendArgs === 'array' ) {
    args = extendArgs;
  }
  else {
    extendArgs = [];
  }

  // do spawn stuff here
}
```

Also, looks like TravisCI has changed some things where `sudo apt-get` doesn't work the same, so that is why the CI is failing. I'll look into fixing this.

File: index.js (hunk 1 above)
Comment (joaomvfsantos) [inline index.js]: Hey there! I think it makes sense. I'll try to refactor this today.

**Comment by joaomvfsantos** [comment]
Hey @ScottONeal any possibility to merge this?

**Comment by ScottONeal** [comment]
merged, I still need to do some changes to get CI to work properly. not NPM publishing until i get that resolved

**Pull request merged**